packages/lib/src/components/internal/PayButton/PayButton.tsx

```diff
@@ -10,27 +10,50 @@ export interface PayButtonProps {
      */
     classNameModifiers?: string[];
     label?: string;
-    amount: PaymentAmount;
+    amount: PaymentAmountExtended;
+    secondaryAmount?: PaymentAmountExtended;
     status?: string;
     disabled?: boolean;
     icon?: string;
     onClick?(): void;
 }
 
-const payAmountLabel = (i18n: Language, amount) =>
-    `${i18n.get('payButton')} ${!!amount?.value && !!amount?.currency ? i18n.amount(amount.value, amount.currency) : ''}`;
+const payAmountLabel = (i18n: Language, amount) => {
+    const payStr = `${i18n.get('payButton')}`;
 
-const PayButton = ({ amount, classNameModifiers = [], label, ...props }: PayButtonProps) => {
+    const primaryAmount =
+        !!amount?.value && !!amount?.currency
+            ? i18n.amount(amount.value, amount.currency, { currencyDisplay: amount.currencyDisplay || 'symbol' })
+            : '';
+
+    return `${payStr} ${primaryAmount}`;
+};
+
+const secondaryAmountLabel = (i18n: Language, secondaryAmount) => {
+    const convertedSecondaryAmount =
+        secondaryAmount && !!secondaryAmount?.value && !!secondaryAmount?.currency
+            ? i18n.amount(secondaryAmount.value, secondaryAmount.currency, { currencyDisplay: secondaryAmount.currencyDisplay || 'symbol' })
+            : '';
+
+    const divider = convertedSecondaryAmount.length ? '/ ' : '';
+
+    return `${divider}${convertedSecondaryAmount}`;
+};
+
+const PayButton = ({ amount, secondaryAmount, classNameModifiers = [], label, ...props }: PayButtonProps) => {
     const { i18n } = useCoreContext();
     const isZeroAuth = amount && {}.hasOwnProperty.call(amount, 'value') && amount.value === 0;
     const defaultLabel = isZeroAuth ? i18n.get('confirmPreauthorization') : payAmountLabel(i18n, amount);
 
+    const secondaryLabel = secondaryAmount && Object.keys(secondaryAmount).length ? secondaryAmountLabel(i18n, secondaryAmount) : null;
+
     return (
         <Button
             {...props}
             disabled={props.disabled || props.status === 'loading'}
             classNameModifiers={[...classNameModifiers, 'pay']}
             label={label || defaultLabel}
+            secondaryLabel={secondaryLabel}
```

Comment:
Just an idea: did you also consider passing `secondaryLabel` as a child component for the Button instead of a prop? 
I think that in this way we can improve the reusability of the Button, as we can pass any sort of custom Components/Values to it.

Then it would be up to the `PayButton` component to pass the secondary label with its specific CSS format, and `Button` doesn't care about that

Ex:
```jsx
<Button...> {secondaryLabel && <SecondaryAmountLabel ... /> </Button>
```

Reply:
You're right - it does make it more powerful... I'll make the change